apps/xmtp.chat/src/content-types/Intent.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  ContentTypeId,
+  type ContentCodec,
+  type EncodedContent,
+} from "@xmtp/content-type-primitives";
+import { z } from "zod";
+
+/**
+ * Intent content structure
+ * Users express their selection by sending Intent messages when they tap action buttons
+ */
+const intentSchema = z.object({
+  /** References Actions.id - provides strong coupling with Actions message */
+  id: z.string(),
+  /** References specific Action.id - indicates which action was selected */
+  actionId: z.string(),
+  /** Optional context data for the selection */
+  metadata: z
+    .record(
+      z.string(),
+      z.union([z.string(), z.number(), z.boolean(), z.null()]),
+    )
+    .optional(),
+});
+
+export type Intent = z.infer<typeof intentSchema>;
+
+/**
+ * Content Type ID for Intent messages
+ * Following XIP-67 specification for inline actions
+ */
+export const ContentTypeIntent = new ContentTypeId({
+  authorityId: "coinbase.com",
+  typeId: "intent",
+  versionMajor: 1,
+  versionMinor: 0,
+});
+
+/**
+ * Intent codec for encoding/decoding Intent messages
+ * Implements XMTP ContentCodec interface for Intent content type
+ */
+export class IntentCodec implements ContentCodec<Intent> {
+  get contentType(): ContentTypeId {
+    return ContentTypeIntent;
+  }
+
+  encode(content: Intent): EncodedContent {
+    // Validate content before encoding
```

Comment:
I don't think that we need the "// Validate content before encoding" comments. At some statements these are present at others they don't. Let's just remove them as the code already reflects that.

Reply:
https://github.com/xmtp/xmtp-js/pull/1219/commits/c1368048e7bfa92ae90dd452faf42aaa41a23b03